src/main/java/com/github/randyklex/dataflow/FilteredLinkPropagator.java

```diff
@@ -70,7 +69,7 @@ public AutoCloseable linkTo(ITargetBlock<T> targetBlock, DataflowLinkOptions lin
 
     public AutoCloseable linkTo(ITargetBlock<T> targetBlock, DataflowLinkOptions linkOptions, Predicate<T> predicate)
     {
-        throw new NotImplementedException();
```

Comment:
This Exception type was pulling in `sun.reflect.generics.reflectiveObjects.NotImplementedException`, which my java installation was having trouble with. After searching around a bit, I found people seem to suggest using `UnsupportedOperationException` for this type of thing, which is part of the default libs.